### ISSUE TITLE
Use secret service in flux

### ISSUE THREAD

**Issue by desa** [influxdata/influxdb#15185]
Flux will need to use the secret service here
https://github.com/influxdata/influxdb/blob/baaf93dbbc8f2b66afeddae47a7ec8e874050763/cmd/influxd/launcher/launcher.go#L472-L488

One thing to note is that the default secret engine is just storing the secrets base64 encoded locally in bolt. This is similar to what k8s does for their default secrets. If the users need something more sophisticated they will be required to use vault.

**Comment by jlapacik** [comment]
This should already happen right here:
https://github.com/influxdata/influxdb/blob/eb15b346ccc8121366364b56e4bd10e046f427d3/cmd/influxd/launcher/launcher.go#L529-L531

**Comment by russorat** [comment]
@jlapacik could you paste a flux script that successfully pulls a secret from bolt for OSS?

**Comment by nisley** [comment]
OSS add secret:
```
curl --data '{"mysqlpassword": "new_value"}' --request PATCH --url http://localhost:9999/api/v2/orgs/039346c3777a1000/secrets --header 'Authorization: Token zx-xrxmnD92Z6gUE-GSlIVqTV0KXxXJFO4_qPapWG-cxkVPaJJwg0A2nicJsNIrotwKPlDDnTbey1Piqof2fZQ=='
```

Query secret:
```
curl --request GET --url http://localhost:9999/api/v2/orgs/039346c3777a1000/secrets --header 'Authorization: Token zx-xrxmnD92Z6gUE-GSlIVqTV0KXxXJFO4_qPapWG-cxkVPaJJwg0A2nicJsNIrotwKPlDDnTbey1Piqof2fZQ=='

{"links":{"org":"/api/v2/orgs/039346c3777a1000","self":"/api/v2/orgs/039346c3777a1000/secrets"},"secrets":["local-postgres-connection","mysqlpassword"]}
```

Flux script (note- only errors when secrets.get line is uncommented):
```
import "sql"

import "influxdata/influxdb/secrets"

pagerduty_secret = secrets.get(key: "mysqlpassword")

from(bucket: "sqlguide")
    |> range(start: -1m)
    |> filter(fn: (r) =>
        (r._measurement == "airSensors" and r._field == "co"))
```

Error message:
```
{"code":"not found","message":"error in evaluating AST while starting program","error":{"code":"not found","message":"error calling function \"get\"","error":{"code":"not found","message":"secret not found"}}}
```

**Comment by jademcgough** [comment]
@nisley has oss flux been updated to include what we need for secrets? Last I heard, some of the flux keywords were only in IDPE.

**Comment by jlapacik** [comment]
After reproducing and subsequently debugging this issue, I discovered that I had associated my secret to the wrong orgID when I wrote to the `/secrets` endpoint. I would have expected the API to not allow me to write a secret for a non-existent org. However after writing to the correct orgID, I was successful in retrieving the secret from bolt using `secrets.get`. I have confirmed this with @nisley. Therefore I am closing this issue.
